index.js

```diff
@@ -1,5 +1,47 @@
 function scoreCalculator(arr) {
-  // do work here
+  // not an array
+  if (!Array.isArray(arr)) {
+    throw new Error('Not an array!');
+  }
+  // not every element of array is a number
+  if (
+    !arr.every(function(elem) {
+      return typeof elem === 'number';
+    })
+  ) {
+    throw new Error('Needs to be an array of numbers!');
+  }
+  // radius scores outside of valid range
+  if (Math.max.apply(null, arr) > 20 || Math.min.apply(null, arr) < 1) {
+    throw new Error('Missed the dartboard! (invalid radius)');
+  }
+
+  if (arr.length === 0) {
+    return -1;
+  }
+
+  function radiusToScore(prev, cur) {
+    let curScore = 0;
+    if (cur >= 5 && cur <= 10) {
+      curScore = 5;
+    } else if (cur >= 1 && cur < 5) {
+      curScore = 10;
+    }
+    return (prev += curScore);
+  }
+
+  let score = arr.reduce(radiusToScore, 0);
+  let bonus = 0;
+
+  // determine whether bonus should be awarded
+  if (
```

Comment:
It's best to do the `.every` outside of an if and place the result into a variable, then test the variable in the if.